<launch>
    <arg name="drone_id" default="1"/>
    <arg name="drone_type" default="x280"/>
    <arg name="racer_x" default="0.0"/>	
    <arg name="racer_y" default="0.0"/>
    <arg name="racer_z" default="0.0"/>


    <!-- 集群管理 -->
    <node pkg="data_manager" type="data_manager" name="data_manager" output="screen" > 
        <rosparam command="load" file="$(find global_interface)/config/fast_lio/mid360.yaml" />
        <remap from="/fastlio_color_pointcloud"   to="fastlio_color_pointcloud"/>
        <remap from="/fastlio_odom"   to="fastlio_odom"/>
        <remap from="/fastlio_path"   to="fastlio_path"/>
        <remap from="/fastlio_pointcloud"   to="fastlio_pointcloud"/>

        <remap from="/global_fastlio_odom"   to="global_fastlio_odom"/>
        <remap from="/global_fastlio_path"   to="global_fastlio_path"/>
        <remap from="/global_fastlio_pointcloud"   to="global_fastlio_pointcloud"/>
        <remap from="odom_mavros"   to="mavros/vision_pose/pose"/>
        <remap from="/odom_racer"   to="/racer_$(arg drone_id)"/>
        <remap from="/pose_racer"   to="/pose_racer_$(arg drone_id)"/>

        <param name="racer_x" value="$(arg racer_x)" type="double"/>
        <param name="racer_y" value="$(arg racer_y)" type="double"/>
        <param name="racer_z" value="$(arg racer_z)" type="double"/>
        <param name="odom_racer_id" value="$(arg drone_id)" type="string"/>
    </node>

    <!-- 集群压缩信息 -->
    <node pkg="data_manager" type="msg_compress_send" name="msg_compress_send" output="screen" > 
    </node>

    <!-- 本地地图维护 -->
    <node name = "local_map" pkg = "data_manager" type = "local_map_node" output = "log">
        <remap from="/cloud_registered"   to="global_fastlio_pointcloud"/>
        <remap from="/global_fastlio_odom"   to="global_fastlio_odom"/>
        <remap from="/local_map"   to="local_map"/>
        <param name="local_x" type="double" value=" 10.0 "/>
        <param name="local_y" type="double" value=" 8.0 "/>
        <param name="local_z" type="double" value=" 3.0 "/>
        <param name="update_x" type="double" value=" 5.0 "/>
        <param name="update_y" type="double" value=" 4.0 "/>
        <param name="update_z" type="double" value=" 3.0 "/>
        <param name="Map_sampling_rate" type="int" value=" 1 "/>
        <param name="downsampleCloud_max" type="double" value=" 0.05 "/>
        <param name="clean_localmap" type="bool" value=" false "/>
    </node>

    <node name = "racer_topic" pkg="data_manager" type="racer_topic" output = "log">
        <remap from="/drone_1/mavros/state" to="/drone_$(arg drone_id)/mavros/state"/>
        <remap from="/drone_1/ctrl_bridge/bridge_status" to="/drone_$(arg drone_id)/ctrl_bridge/bridge_status"/>
    </node>


</launch>
